packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarRing.tsx

```diff
@@ -0,0 +1,81 @@
+import { makeStyles } from '@masknet/theme'
+import { RainbowBox } from './RainbowBox'
+
+const useStyles = makeStyles()((theme) => ({
+    root: {
+        overflow: 'unset',
+    },
+}))
+interface NFTAvatarRingProps {
+    stroke: string
+    strokeWidth: number
+    fontSize: number
+    text: string
+    width: number
+}
+
+export function NFTAvatarRing(props: NFTAvatarRingProps) {
+    const { classes } = useStyles()
+    const { stroke, strokeWidth, fontSize, text, width } = props
+
+    const R = width + 2 * strokeWidth - 4
+    const r = R / 2 - strokeWidth
+    const path_r = R / 2 - strokeWidth + fontSize / 2
+    const x1 = R / 2 - path_r / 2
+    const y1 = R / 2 + Math.sqrt(Math.pow(path_r, 2) - Math.pow(path_r / 2, 2))
+    const x2 = R / 2 + path_r / 2
+    const y2 = y1
+
+    return (
+        <RainbowBox width={R} height={R}>
+            <svg className={classes.root} width={R} height={R} viewBox={`0 0 ${R} ${R}`} id="NFTAvatarRingIcon">
+                <defs>
+                    <path
+                        id="path"
+                        fill="none"
+                        stroke="none"
+                        strokeWidth="0"
+                        d={`M${x1} ${y1} A${path_r} ${path_r} 0 1 1 ${x2} ${y2}`}
+                    />
+                    <linearGradient id="gradient" x1="0%" y1="0%" x2="100%" y2="0">
```

Comment:
Create an uniq id for every rendering

```suggestion
                    <linearGradient id={uniqId} x1="0%" y1="0%" x2="100%" y2="0">
```